### ISSUE TITLE
CI: K8sServicesTest Checks service across nodes Tests NodePort (kube-proxy) with externalTrafficPolicy=Local

### ISSUE THREAD

**Issue by pchaigno** [cilium/cilium#15166]
The two following tests of externalTrafficPolicy=Local with kube-proxy are failing 100% of the time on master when run with K8s 1.14 and a 4.9 kernel:
```
Suite-k8s-1.14.K8sServicesTest Checks service across nodes Tests NodePort (kube-proxy) with externalTrafficPolicy=Local
Suite-k8s-1.14.K8sServicesTest Checks service across nodes Tests NodePort (kube-proxy) with the host firewall and externalTrafficPolicy=Local
```
https://jenkins.cilium.io/job/cilium-master-k8s-1.14-kernel-4.9/14/#showFailuresLink
The artifacts are missing because of https://github.com/cilium/cilium/issues/15128.

### Stacktrace
```
/home/jenkins/workspace/cilium-master-k8s-1.14-kernel-4.9/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:514
Request from k8s2 to service http://192.168.36.11:30202 failed
Expected command: kubectl exec -n kube-system log-gatherer-69hp7 -- /bin/bash -c 'fails=""; id=$RANDOM; for i in $(seq 1 10); do if curl --path-as-is -s -D /dev/stderr --fail --connect-timeout 5 --max-time 20 http://192.168.36.11:30202 -H "User-Agent: cilium-test-$id/$i"; then echo "Test round $id/$i exit code: $?"; else fails=$fails:$id/$i=$?; fi; done; if [ -n "$fails" ]; then echo "failed: $fails"; fi; cnt="${fails//[^:]}"; if [ ${#cnt} -gt 0 ]; then exit 42; fi' 
To succeed, but it failed:
Exitcode: 42 
Err: exit status 42
Stdout:
 	 failed: :22913/1=28:22913/2=28:22913/3=28:22913/4=28:22913/5=28:22913/6=28:22913/7=28:22913/8=28:22913/9=28:22913/10=28
	 
Stderr:
 	 command terminated with exit code 42
```

https://datastudio.google.com/s/uLteFjIRe0k shows that most runs of these tests on K8s 1.14 are failing. The only passing tests are for backport PRs to v1.8 and v1.9 (cf. https://datastudio.google.com/s/kFBoCFVs08o).

https://github.com/cilium/cilium/issues/15103 seems related but may not be the same issue as the present failure is happening 100% of the time and is clearly specific to K8s 1.14.

**Comment by nbusseneau** [comment]
Quarantined by #15415.

**Comment by nbusseneau** [comment]
Since K8s 1.14 is being dropped in #15502, we should revert #15415 after it is merged, closing this in the process.

**Comment by nbusseneau** [comment]
Fixed by #15502: K8s 1.14 support was dropped, and the PR directly reverted #15415 changes as well.
